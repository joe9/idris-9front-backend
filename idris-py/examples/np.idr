import Data.Vect
import Python
import Python.Lib.Numpy.Matrix

%default total

f : Nat -> Maybe (List Bool)
f Z = Just [True, False]
f (S n) = f n

xs : Matrix 3 4 DDouble
xs = array _
  [[1.0,-2.1, 3.3, -0.1]
  ,[3.5, 7.0, 0.0, -5.2]
  ,[0.5, 7.2,-1.1,  0.0]
  ]

ys : Matrix 6 2 DDouble
ys = reshape xs

zs : Matrix 4 3 DDouble
zs = reshape ys

main : PIO ()
main = do
  printLn' $ f 4
  printLn' xs
  printLn' ys
  printLn' zs
  -- printLn (ys `dot` zs)  -- won't work!
  printLn' (xs `dot` zs)
  printLn' (xs `dot` transpose xs)
  printLn' (transpose xs `dot` xs)
  printLn' $ 2 * transpose (zs `dot` xs) + fromDouble 0.2
