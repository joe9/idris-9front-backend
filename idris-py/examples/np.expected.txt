Just [True, False]
[[ 1.  -2.1  3.3 -0.1]
 [ 3.5  7.   0.  -5.2]
 [ 0.5  7.2 -1.1  0. ]]
[[ 1.  -2.1]
 [ 3.3 -0.1]
 [ 3.5  7. ]
 [ 0.  -5.2]
 [ 0.5  7.2]
 [-1.1  0. ]]
[[ 1.  -2.1  3.3]
 [-0.1  3.5  7. ]
 [ 0.  -5.2  0.5]
 [ 7.2 -1.1  0. ]]
[[  0.49 -26.5   -9.75]
 [-34.64  22.87  60.55]
 [ -0.22  29.87  51.5 ]]
[[ 16.31 -10.68 -18.25]
 [-10.68  88.29  52.15]
 [-18.25  52.15  53.3 ]]
[[  13.5    26.      2.75  -18.3 ]
 [  26.    105.25  -14.85  -36.19]
 [   2.75  -14.85   12.1    -0.33]
 [ -18.3   -36.19   -0.33   27.05]]
[[  -9.2    31.5   -35.7     6.9 ]
 [  14.12  150.42  -65.4   -45.44]
 [  -0.46  -15.86   -0.9    47.72]
 [  21.84  -36.18   54.28   10.2 ]]
